src/utils/user_registry.ts

```diff
@@ -3,124 +3,81 @@ import {
   IResponseErrorInternal,
   IResponseErrorValidation,
   ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
   ResponseErrorValidation,
 } from "@pagopa/ts-commons/lib/responses";
-import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
 import { toError } from "fp-ts/lib/Either";
+import { Option, some } from "fp-ts/lib/Option";
+import { SaveUserDto } from "../../generated/pdv-userregistry-api/SaveUserDto";
+import { UserId } from "../../generated/pdv-userregistry-api/UserId";
 import { pipe } from "fp-ts/lib/function";
+
 import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
 import * as O from "fp-ts/lib/Option";
+import { PersonalDatavaultAPIClient } from "../clients/pdv_client";
+import { toResponseErrorInternal } from "./conversions";
 
-import * as TE from "fp-ts/lib/TaskEither";
-import { none, Option, some } from "fp-ts/lib/Option";
-import { User } from "../../generated/userregistry-api/User";
-import { UserSeed } from "../../generated/userregistry-api/UserSeed";
+type ErrorResponses =
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+  | IResponseErrorValidation;
 
-import { UserRegistryAPIClient } from "../clients/userregistry_client";
-import { errorsToError, toResponseErrorInternal } from "./conversions";
-import { logger } from "./logger";
-export const getUserId = (
-  apiClient: ReturnType<UserRegistryAPIClient>,
-  externalId: FiscalCode,
+export const blurUser = (
+  pdvClient: ReturnType<PersonalDatavaultAPIClient>,
+  user: SaveUserDto,
   subscriptionKey: NonEmptyString
-): TE.TaskEither<
-  IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized,
-  Option<{ id: string }>
-> =>
+): TE.TaskEither<ErrorResponses, Option<Pick<UserId, "id">>> =>
```

Comment:
Is there actually a scenario in which we get `Right<None>`? As I see, if the value is nullable we always return a `Left<IResponseErrorInternal>` with `Error reading response data`.

I think we can simply have a result of type 
```
TE.TaskEither<ErrorResponses, Pick<UserId, "id">>
```

Reply:
@balanza solved by https://github.com/pagopa/hub-spid-login-ms/pull/52/commits/b21092844d574e5b6afdb818d8982e6b037980be